app/packs/src/fetchers/CollectionsFetcher.js

```diff
@@ -49,6 +49,96 @@ export default class CollectionsFetcher {
     return promise;
   }
```

Comment:
fetchMyRoots( ) , fetchSharedWithMeRoots ( ), fetchMyRoots ( ), fetchSharedWithMeRoots( ), fetchMyRoots ( ) and fetchSharedWithMeRoots ( ) all have the same structure/logic. Only the adress is different. I would extract the structure and have general api caller method with the address as a parameter